packages/@vue/cli-service/generator/template/public/index.html

```diff
@@ -5,11 +5,11 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width,initial-scale=1.0">
     <link rel="icon" href="<%%= BASE_URL %%>favicon.ico">
-    <title><%= rootOptions.projectName %></title>
+    <title><%%= htmlWebpackPlugin.options.title %%></title>
   </head>
   <body>
     <noscript>
-      <strong>We're sorry but <%= rootOptions.projectName %> doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
+      <strong>We're sorry but <%%= htmlWebpackPlugin.options.title %%> doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
```

Comment:
Why can't we use `projectName` from `vue.config.js`?

Reply:
`rootOptions.projectName` is inferred from `package.json` so it's equivalent to the default `title` option value set below.

This PR is to make the `title` element in the template **dynamic**, so that users can change the page title by configuring `html-webpack-plugin`. Most common use cases may be in multi-page apps.

I originally thought the `title` option would be injected into the page by default, like `meta` tags. But it turns out that users need to place an interpolation expression in the template manually. https://github.com/jantimon/html-webpack-plugin/issues/1302#issuecomment-551369134